jupyter_server/gateway/gateway_client.py

```diff
@@ -275,29 +345,22 @@ def validate_cert_default(self):
             not in ["no", "false"]
         )
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self._static_args = {}  # initialized on first use
-
-        # store of cookies with store time
-        self._cookies = {}  # type: ty.Dict[str, ty.Tuple[Morsel, datetime]]
-
     allowed_envs_default_value = ""
     allowed_envs_env = "JUPYTER_GATEWAY_ALLOWED_ENVS"
     allowed_envs = Unicode(
         default_value=allowed_envs_default_value,
         config=True,
         help="""A comma-separated list of environment variable names that will be included, along with
-         their values, in the kernel startup request.  The corresponding `allowed_envs` configuration
-         value must also be set on the Gateway server - since that configuration value indicates which
-         environmental values to make available to the kernel. (JUPYTER_GATEWAY_ALLOWED_ENVS env var)""",
+their values, in the kernel startup request.  The corresponding `client_envs` configuration
+value must also be set on the Gateway server - since that configuration value indicates which
+environmental values to make available to the kernel. (JUPYTER_GATEWAY_ALLOWED_ENVS env var)""",
     )
 
     @default("allowed_envs")
     def _allowed_envs_default(self):
         return os.environ.get(
-            "JUPYTER_GATEWAY_ENV_WHITELIST",
-            os.environ.get(self.allowed_envs_env, self.allowed_envs_default_value),
+            self.allowed_envs_env,
+            os.environ.get("JUPYTER_GATEWAY_ENV_WHITELIST", self.allowed_envs_default_value),
```

Comment:
```suggestion
            os.environ.get("JUPYTER_GATEWAY_ALLOWED_ENVS", self.allowed_envs_default_value),
```

Reply:
There's an extraneous trailing comma, but the idea here is that users could still be using an env of `JUPYTER_GATEWAY_ENV_WHITELIST` that should be used unless they have defined env `JUPYTER_GATEWAY_ALLOWED_ENVS`, in which case, that should override.  If neither env is set, then `self.allowed_envs_default_value` is used.

Previously, I had the two envs switched, which would honor the `ENV_WHITELIST` env over `ALLOWED_ENVS`, but we should prefer the latter.

Does that make sense?  (I'll hold off on the commit to remove the comma until we decide the best way to move forward here.)

Reply:
Ah, right, that makes sense.

Reply:
@blink1073 - Turns out pre-commit (black) wants that "extraneous" comma in there (and local testing appears to be fine), so there won't be another commit to remove it. :smile: